### PR TITLE
[Serve] Fixed `_DeploymentHandleBase` to avoid regenerating `handle` tag every time `handle.options` is invoked

### LINKED ISSUE #41282: [Serve] Specifying `DeploymentHandle.options` generates new Deployment id in metrics, blowing up cardinality
### What happened + What you expected to happen

Following Serve examples of how to use it leads to an issue where for every request new `DeploymentHandle` is generated, bearing new id, therefore producing new time-series for every metric emitted by it.

This severely blows up metrics cardinality putting unnecessary stress on the metrics ingestion pipeline.

Since following use-case is recommended way of using DH, we'd make sure when we create a new one we reuse the id to avoid generating a new for every request made:

```
response: DeploymentResponse = handle.options(
                method_name="other_method",
                multiplexed_model_id="model:v1",
                stream=True,
            ).remote()
```

### Versions / Dependencies

Ray 2.8

### Reproduction script

```
handle = Deployment.bind(...)
# This will create a new DH with new id
handle.options(stream=True).rpc.remote()
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by alexeykudinkin**
## Why are these changes needed?

Addresses https://github.com/ray-project/ray/issues/41282

This makes sure that since we're reusing the counter we also don't update its default tags set

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
LGTM!

File: python/ray/serve/tests/test_handle.py [hunk 1]
```diff
@@ -186,16 +186,21 @@ def __call__(self):
...
+    counter = handle1.request_counter
+    counter_info = counter.info
     assert handle1.remote().result() == "__call__"
 
     handle2 = handle1.options(method_name="method_a")
+
     assert handle2.remote().result() == "method_a"
-    assert handle2.request_counter == metrics
+    assert handle2.request_counter == counter
+    assert handle2.request_counter.info == counter_info
```
Comment (alexeykudinkin) [inline python/ray/serve/tests/test_handle.py:197]: This should catch default tags being overridden previously

Comment (GeneDer) [reply python/ray/serve/tests/test_handle.py:197]: Thanks for adding the test!

**Review by GeneDer — APPROVED** [review]

**Comment by alexeykudinkin** [comment]
LinkCheck is failing with:

```

-rate limited-   https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8 \| sleeping...
--
  | -rate limited-   https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205 \| sleeping...
  | -rate limited-   https://medium.com/distributed-computing-with-ray/machine-learning-serving-is-broken-f59aff2d607f \| sleeping...
  | (ray-core/examples/map_reduce: line 180002) broken    https://medium.com/distributed-computing-with-ray/executing-adistributed-shuffle-without-a-mapreduce-system-d5856379426c - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/executing-adistributed-shuffle-without-a-mapreduce-system-d5856379426c


```

**Pull request merged**